### PR TITLE
Use error handler for reactive cache aspect

### PR DESCRIPTION
**Pull request by zizare**
Use cacheErrorHandler on reactive caching aspect.

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java [hunk 1]
```diff
@@ -506,9 +506,12 @@ private Object findInCaches(CacheOperationContext context, Object key,
 
 		for (Cache cache : context.getCaches()) {
 			if (CompletableFuture.class.isAssignableFrom(context.getMethod().getReturnType())) {
-				CompletableFuture<?> result = cache.retrieve(key);
+					CompletableFuture<?> result = cache.retrieve(key);
```
Comment (simonbasle) [inline spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java:509]: nit: formatting left-over from the 1st commit

File: spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java [hunk 2]
```diff
@@ -1136,12 +1139,30 @@ public Object findInCaches(CacheOperationContext context, Cache cache, Object ke
 				if (adapter.isMultiValue()) {
 					return adapter.fromPublisher(Flux.from(Mono.fromFuture(cachedFuture))
 							.switchIfEmpty(Flux.defer(() -> (Flux) evaluate(null, invoker, method, contexts)))
-							.flatMap(v -> evaluate(valueToFlux(v, contexts), invoker, method, contexts)));
+							.flatMap(v -> evaluate(valueToFlux(v, contexts), invoker, method, contexts))
+							.onErrorResume(RuntimeException.class, ex -> {
+								try {
+									getErrorHandler().handleCacheGetError((RuntimeException) ex, cache, key);
+									return Flux.defer(() -> (Flux) evaluate(null, invoker, method, contexts));
```
Comment (simonbasle) [inline spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java:1146]: I'm not entirely convinced that the `defer` is necessary here (similarly to how there's no defer in the `flatMap` above). was there a particular reason for it?

Comment (zizare) [reply spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java]: You right it's not necessary here.
Removed it

**Comment by simonbasle** [comment]
Thanks for the contribution @zizare 👍 this will be released in 6.1.11 and 6.2.0-M5, both coming up on July 11

**Pull request closed without merging**